identity_storage/src/storage/tests/presentation_validation.rs

```diff
@@ -91,6 +92,51 @@ where
   );
 }
 
+#[tokio::test]
+async fn test_extract_dids() {
+  test_extract_dids_impl(setup_coredocument(None, None).await).await;
+  test_extract_dids_impl(setup_iotadocument(None, None).await).await;
+}
+async fn test_extract_dids_impl<T>(setup: Setup<T, T>)
+where
+  T: JwkDocumentExt + AsRef<CoreDocument>,
+{
+  let credential: CredentialSetup = generate_credential(&setup.issuer_doc, &[&setup.subject_doc], None, None);
+  let jws = sign_credential(&setup, &credential.credential).await;
+
+  let presentation: JwtPresentation =
+    JwtPresentationBuilder::new(setup.subject_doc.as_ref().id().to_url().into(), Object::new())
+      .credential(jws)
+      .build()
+      .unwrap();
```

Comment:
I don't think we have a test case where we add multiple credentials to a presentation yet and this would be a good place to test it.